### PR TITLE
miscellaneous doc improvements

### LINKED ISSUE #49: For Documentation: auto-cutoff? and consequences?
In the [documentation](https://zero-to-jupyterhub-with-kubernetes.readthedocs.io/en/latest/create-k8s-cluster.html), it would be helpful to have information about whether one can put an automatic limit on how much their cloud account can be billed.  Also, if that feature is available, what would the consequences be?  What would happen to the system when the shutoff is triggered?

### LINKED ISSUE #58: Helm question
Unclear to me:

Should helm  be installed once, or per cluster?

While I can figure it out, it could be made clearer in the docs. 
Also you could separate it with "all you need to install on your local machine" on a first section.

### LINKED ISSUE #64: Document how to recover from a failed install
Need to delete namespace probably. Easiest way to do this.

### LINKED ISSUE #93: Include a "things we'd love help with" section
What do you all think about including a section listing some of the high-level things we need help with (e.g., Julia support in repo2docker, Azure / XSEDE support)

### LINKED ISSUE #131: Ambiguous instructions in helm setup
Where is this line supposed to be executed? 

https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blame/master/doc/source/setup-helm.rst#L31

Are you supposed to log into your cluster to do this? Into the head node? Into each of the worker nodes? Or do you do this on your local machine?

### LINKED ISSUE #150: add gcloud free tier limitations
When mentioning the free credits here: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/doc/source/create-k8s-cluster.rst

We should also mention the free tier limitations (which limits the total number of users the cluster can support using purely the free $300 credits, depending on the configs in the `config.yaml`).

https://cloud.google.com/free/docs/frequently-asked-questions#limitations

> What limitations does the free trial have?
> 
> Google Compute Engine limitations
> 
> During the free trial, your project can have no more than 8 cores (or virtual CPUs) running at the same time.
> 
> For example, you can launch eight n1-standard-1 machines, or two n1-standard-4 machines, but you cannot launch a n1-standard-16 machine because the machine would have 16 vCPUs, which is more than 8 vCPUs. Similarly, you cannot launch an n1-standard-1 and an n1-standard-8 machine, because that would mean 9 vCPUs in the project. To run more than 8 vCPUs at a time, you must upgrade your account.
> 
> For more information about the types of virtual machines available and the number of cores they use, see Machine type pricing.

https://cloud.google.com/container-engine/pricing#a_note_about_kubernetes_master_nodes

> 
> A note about Kubernetes master nodes
> 
> If you're coming to us from the open-source Kubernetes project, you're used to the master node running on an additional machine instance. Google Container Engine abstracts away the existence of the master; it doesn't count as an instance for the purpose of pricing.

### LINKED ISSUE #414: Document how to let users create / activate custom conda environments
From https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/393 we figured out how to allow this. We should document this (maybe just in `FAQ`?) somewhere. This comment and the couple below it show the magic incantation to make this happen:

https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/393#issuecomment-358108096

### LINKED ISSUE #446: {{ possible security issue }} -> Document security reporting process
{{ possible security issue }}

### PR DESCRIPTION
**Pull request by choldgraf**
This is several documentation improvements for a bunch of issues that we've been collecting over time. Spot-checks a number of questions / concerns / suggestions that people have raised. Here's a list of all issues this will address:

closes #93
closes #414
closes #446
closes #64
closes #49
closes #131
closes #150
closes #58

### CODE REVIEW

File: doc/source/google/step-zero-gcp.rst [hunk 1]
```diff
@@ -71,6 +74,12 @@ connect your credit card or other payment method to your google cloud account.
    * ``--zone`` specifies which data center to use. Pick something that is not
      too far away from your users. You can find a list of them `here <https://cloud.google.com/compute/docs/regions-zones/regions-zones#available>`_.
 
+   .. note::
+
+      Consider `setting a cloud budget <https://cloud.google.com/billing/docs/how-to/budgets>`_
+      for your Google Cloud account in order to make sure you don't accidentally
+      spend more than you wish to.
+
```
Comment (consideRatio) [inline doc/source/google/step-zero-gcp.rst]: Wieee it existed for GCP as well!

File: doc/source/security.md [hunk 2]
```diff
@@ -4,6 +4,16 @@ The information in this document focuses primarily on cloud based deployments. F
...
+## Reporting a security issue
+
+If you find a security vulnerability in JupyterHub, either a failure of the
+code to properly implement the model described here, or a failure of the
+model itself, please report it to [security@ipython.org](mailto:security@ipython.org).
+
+If you prefer to encrypt your security reports, you can use
+[this PGP public key](https://jupyter-notebook.readthedocs.io/en/stable/_downloads/ipython_security.asc).
+
+
```
Comment (consideRatio) [inline doc/source/security.md]: Excellent!

File: doc/source/setup-jupyterhub.rst [hunk 3]
```diff
@@ -120,7 +120,11 @@ Install JupyterHub
         ``helm delete --purge <YOUR-RELEASE-NAME>``. Then reinstall by repeating this
         step. If it persists, also do ``kubectl delete <YOUR-NAMESPACE>`` and try again.
 
-        If you're pulling from a large Docker image you may get a
+      * In general, if something goes *wrong* with the install step, delete the
+        Helm namespace by running ````helm delete --purge <YOUR-RELEASE-NAME>``
```
Comment (consideRatio) [inline doc/source/setup-jupyterhub.rst]: \`\` typo, right?

File: doc/source/user-environment.rst [hunk 4]
```diff
@@ -345,6 +347,28 @@ for more information on using this tool.
...
+   sure that the following lines are inside:
+
+   ```
+   envs_dirs:
+     - /home/jovyan/my-conda-envs/
+   ```
+
+   The text above will cause Anaconda to install new environments to this
+   folder, which will persist across sessions.
+
```
Comment (consideRatio) [inline doc/source/user-environment.rst]: Ohhhhh.... Another excellent addition!

**Comment by choldgraf** [comment]
updated with @consideRatio 's typo fix! plus a bit of languages to address helm initialization from #58!

File: doc/source/setup-helm.rst [hunk 5]
```diff
@@ -59,6 +64,22 @@ shell from your provider), enter:
...
+   The local and remote version of ``helm`` must be the same in order to
+   ensure they can talk to each other. If you wish to run ``helm`` commands
+   from a *new* computer than the one used to run the commands above, you
+   must re-initialize it by running the following modified version of
+   the ``init`` command:
+
+   ``helm init --client-only --service-account tiller``
+
+   This will initialize ``helm`` locally, according to the version that is
+   running remotely on the cluster. Note that this requires ``kubectl``
```
Comment (manics) [inline doc/source/setup-helm.rst]: With `--client-only` I'm pretty sure helm won't connect to the cluster, and you have to download the correct version.

Comment (choldgraf) [reply doc/source/setup-helm.rst]: ah no, you're probably right, I was just optimistically assuming :-)

Comment (choldgraf) [reply doc/source/setup-helm.rst]: that said, any reason *not* to just run `helm init` again? I guess if your local helm has a different version maybe it'll update the cluster version and invalidate other people's helm connection?

Comment (manics) [reply doc/source/setup-helm.rst]: Yes, that's a good reason not to. If someone is new to z2jh they're probably testing on their own from a single computer so this isn't relevant, and if multiple people/computers are involved it's probably going beyond just a test installation so I think it's better to be safe.

File: doc/source/setup-helm.rst [hunk 6]
```diff
@@ -58,6 +64,22 @@ cluster. At the terminal, enter:
 
 This command only needs to run once per Kubernetes cluster.
 
+.. note::
+
+   The local and remote version of ``helm`` must be the same in order to
+   ensure they can talk to each other. If you wish to run ``helm`` commands
+   from a *new* computer than the one used to run the commands above, you
```
Comment (willingc) [inline doc/source/setup-helm.rst]: How about *different* instead of *new*? Or a second computer.

File: doc/source/setup-helm.rst (hunk 5 above)
Comment (willingc) [reply doc/source/setup-helm.rst]: I would leave it as written. Perhaps remind again `helm version` will output the version of the client and the server.

File: doc/source/security.md (hunk 2 above)
Comment (jzf2101) [reply doc/source/security.md]: This should probably go in the contributing or the readme too

File: doc/source/setup-helm.rst (hunk 6 above)
Comment (jzf2101) [reply doc/source/setup-helm.rst]: I agree

**Pull request merged**